src/toil/cwl/cwltoil.py

```diff
@@ -3598,20 +3809,23 @@ def visitSteps(
         # For workflows we need to dispatch on steps
         for step in cmdline_tool.steps:
             # Handle the step's tool
-            op(step.tool)
+            ret = op(step.tool)
             # Recures on the embedded tool; maybe it's a workflow.
-            visitSteps(step.embedded_tool, op)
+            recurse_ret = visitSteps(step.embedded_tool, op)
+            ret.extend(recurse_ret)
+            return ret
```

Comment:
@stxue1 I'm not sure that this returns a list?  Is this intended to always return the first (and only the first) step?  Maybe this shouldn't be a loop if so, and `cmdline_tool.steps[0]` used, just to make it clear.

Reply:
To be honest I'm not too certain why this is done in a loop and returned after the first step. `cmdline_tool.steps[0]` would probably work too, except it seems like it can be possible for the cmdline_tool to have no steps and that we should error in that case. As long as the `cmdline_tool` has steps though, the funciton should always return a list though

Reply:
I think I forgot to visit all steps, I moved the return statement down so it should now return the *right* list